python-package/lightgbm/basic.py

```diff
@@ -594,6 +598,75 @@ def _load_pandas_categorical(file_name=None, model_str=None):
         return None
 
 
+class Sequence(object):
```

Comment:
```suggestion
class Sequence:
```

Since LightGBM no longer supports Python 2 (https://github.com/microsoft/LightGBM/pull/4089#discussion_r620355287), it isn't necessary to explicitly declare that a class inherits from `object`. We have a preference for not using `(object)` in class declarations. For example, see these other class declarations in this file:

https://github.com/microsoft/LightGBM/blob/32fec820607d8cbcd1fe37c200455ea45702df77/python-package/lightgbm/basic.py#L1047

https://github.com/microsoft/LightGBM/blob/32fec820607d8cbcd1fe37c200455ea45702df77/python-package/lightgbm/basic.py#L2150